packages/@ember/-internals/metal/lib/chain-tags.ts

```diff
@@ -17,46 +17,63 @@ export function finishLazyChains(obj: any, key: string, value: any) {
   }
 
   if (value === null || (typeof value !== 'object' && typeof value !== 'function')) {
-    lazyTags.length = 0;
+    for (let path in lazyTags) {
+      delete lazyTags[path];
```

Comment:
Is `lazyTags` a dictionary?

Reply:
Yes, created with `Object.create(null)`. That was one of the optimizations I made. The overall difference in perf was very small, but seemed positive, so I kept it. We can drop it and go back to the 2d array as well.